### PR TITLE
feature: api tests setup

### PR DESCRIPTION
**Pull request by lucasmenendez**
* Testing API setup includes Voconed, Vocdoni Faucet, MongoDB and API service.
* MongoDB instance is globally available to be accessed directly in unit tests.
* Initial register endpoint tests done.

### CODE REVIEW

File: api/api_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+// testURL helper function returns the full URL for the given path using the
+// test host and port.
+func testURL(path string) string {
+	return fmt.Sprintf("http://%s:%d%s", testHost, testPort, path)
+}
+
+// mustMarshall helper function marshalls the input interface into a byte slice.
+// It panics if the marshalling fails.
+func mustMarshall(i any) []byte {
```
Comment (altergui) [inline api/api_test.go:46]: mustMarshal

File: api/users_test.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	"bytes"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+)
+
+func Test_registerHandler(t *testing.T) {
```
Comment (altergui) [inline api/users_test.go:13]: idiomatic would be TestRegisterHandler but i guess you have a reason to deviate

**Review by altergui — APPROVED** [review]
LGTM

File: api/users_test.go (hunk 2 above)
Comment (lucasmenendez) [reply api/users_test.go:13]: I thought that if the method to test is private the test function must follow that format, but reading again about this I was wrong :(

**Pull request merged**